src/Neo/Extensions/NeoSystemExtensions.cs

```diff
@@ -0,0 +1,90 @@
+// Copyright (C) 2015-2025 The Neo Project.
+//
+// NeoSystemExtensions.cs file belongs to the neo project and is free
+// software distributed under the MIT software license, see the
+// accompanying file LICENSE in the main directory of the
+// repository or http://www.opensource.org/licenses/mit-license.php
+// for more details.
+//
+// Redistribution and use in source and binary forms with or without
+// modifications are permitted.
+
+/* Unmerged change from project 'Neo(net9.0)'
+Before:
+using System;
+After:
+// Copyright (C) 2015-2025 The Neo Project.
+//
+// NeoSystemExtensions.cs file belongs to the neo project and is free
+// software distributed under the MIT software license, see the
+// accompanying file LICENSE in the main directory of the
+// repository or http://www.opensource.org/licenses/mit-license.php
+// for more details.
+//
+// Redistribution and use in source and binary forms with or without
+// modifications are permitted.
+
+using System;
+*/
+// Copyright (C) 2015-2025 The Neo Project.
+//
+// NeoSystemExtensions.cs file belongs to the neo project and is free
+// software distributed under the MIT software license, see the
+// accompanying file LICENSE in the main directory of the
+// repository or http://www.opensource.org/licenses/mit-license.php
+// for more details.
+//
+// Redistribution and use in source and binary forms with or without
+// modifications are permitted.
+
+using Neo.Network.P2P.Payloads;
+using Neo.Persistence;
+using Neo.SmartContract.Native;
+using System;
+
+namespace Neo
+{
+    /// <summary>
+    /// Extension methods for <see cref="NeoSystem"/> and related types.
+    /// </summary>
+    public static class NeoSystemExtensions
+    {
+        /// <summary>
+        /// Gets the block generation time based on the current state of the blockchain.
+        /// </summary>
+        /// <param name="system">The NeoSystem instance.</param>
+        /// <returns>The block generation time as a TimeSpan.</returns>
+        public static TimeSpan GetBlockGenTime(this NeoSystem system)
+        {
+            return system.StoreView.GetBlockGenTime(system.Settings);
+        }
+
+        /// <summary>
+        /// Gets the block generation time based on the current state of the blockchain.
+        /// </summary>
+        /// <param name="snapshot">The snapshot of the store.</param>
+        /// <param name="settings">The protocol settings.</param>
+        /// <returns>The block generation time as a TimeSpan.</returns>
+        public static TimeSpan GetBlockGenTime(this IReadOnlyStore snapshot, ProtocolSettings settings)
+        {
+            try
+            {
+                // Get the current block height from the blockchain
+                var index = NativeContract.Ledger.CurrentIndex(snapshot);
+
+                // Before the Echidna hardfork, use the protocol settings
+                if (!settings.IsHardforkEnabled(Hardfork.HF_Echidna, index))
```

Comment:
After the hardfork is implemented I think we should remove this and set a default 15s on the storage itself.

Reply:
It is ok like this for now anyway